### PR TITLE
Initialized FormGroup Component

### LINKED ISSUE #669: 🔮 [New Component] FormGroup
Add a new `FormGroup` component which adds styles for multiple form elements used together.

https://primer.style/css/components/forms#form-groups

### PR DESCRIPTION
**Pull request by haigdouzdjian**
Created component for Form Group

Closes #669 

### Screenshots
Screen Shot 2020-07-20 at 9 16 28 PM

### Merge checklist
- [x] Added or updated TypeScript definitions (`index.d.ts`) if necessary
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

### CODE REVIEW

**Comment by colebemis** [comment]
Hi, @haigdouzdjian 👋 Thanks for opening this PR!

It looks like you've hard-coded a lot of values in the `FormGroup` component that consumers might want to customize (e.g. label and placeholder text). 

We could allow more customization by adding extra props to `FormGroup`, but a more flexible approach would be to let the consumer handle the rendering of the form controls. Here's the API I would propose:

### Before

```jsx
<FormGroup variant="input" />
```

### After

```jsx
<FormGroup>
  <FormGroup.Label htmlFor="example-text">Example text</FormGroup.Label>
  <TextInput id="example-text" value="Example Value" />
</FormGroup>
```

An implementation for this API would look something like this:

```js
// FormGroup.js

// ... imports ... 

const FormGroup = styled.div`
  margin-top: ${get('space.3')};
  margin-bottom: ${get('space.3')};
  ${COMMON};
  ${sx};
`

FormGroup.defaultProps = {theme}

FormGroup.propTypes = {
  ...COMMON.propTypes,
  ...sx.propTypes
}

FormGroup.Label = styled.label`
  display: block;
  margin-bottom: ${get('space.1')};
  font-size: ${get('fontSizes.1')};
  font-weight: ${get('fontWeights.bold')};
  ${COMMON};
  ${TYPOGRAPHY};
  ${sx};
`
FormGroup.Label.defaultProps = {theme}

FormGroup.Label.propTypes = {
  ...COMMON.propTypes,
  ...TYPOGRAPHY.propTypes,
  ...sx.propTypes
}

export default FormGroup
```

File: docs/content/FormGroup.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+---
+
+Adds styles for multiple form elements used together.
+
+## Default example
+
+```jsx live
+<FormGroup>
+  <FormGroup.Label htmlFor="example-text">Example text</FormGroup.Label>
+  <TextInput id="example-text" width="440px" bg="#fafbfc" value="Example Value" />
```
Comment (colebemis) [inline docs/content/FormGroup.md:12]: I don't think we need these props in the example

```suggestion
  <TextInput id="example-text" value="Example Value" />
```

**Comment by colebemis** [comment]
Oops, forgot to mention that you'll need to update snapshot tests after making styling changes.

To update the snapshots, run `yarn update-snapshots` in your terminal. That should fix the failing tests :)

File: src/FormGroup.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import {COMMON, get, TYPOGRAPHY} from './constants'
+import theme from './theme'
+import sx from './sx'
+
+const FormGroup = styled.div`
+  margin: ${get('space.3')} 0;
+  font-weight: ${get('fontWeights.normal')};
```
Comment (emplums) [inline src/FormGroup.js:9]: It looks like the FormGroup CSS in Primer CSS adds some additional styles to text inputs:
image

I think we'll want everything above added here - I would probably do this by targeting the `input` as a nested selected here on line 10.

Comment (emplums) [reply src/FormGroup.js:9]: It looks like there are also some styles specific to text areas that are included in Primer CSS's form group styles - we should add those here as well!

**Review by emplums — APPROVED** [review]
Hey @haigdouzdjian thanks for this PR! I left a few comments, but feel free to merge this as is and I can follow up with adding those extra styles in a separate PR! This looks great, thanks for the contribution! :)

**Pull request merged**